Create inverter board abstraction to send velocity and throttle to Raspberry Pi Pico

samderanova opened the pull request:
Resolves #35.

## Changes
- Add new `InverterBoard` struct
- Write `send_control` method to send a velocity and throttle to a Raspberry Pi Pico via UART serial communication over `/dev/ttyACM0` (i.e. a USB Type A port on the Raspberry Pi)
- Include demo function that sends a velocity of 0 and a throttle of 1 to a Pico every second

## Testing
1. Plug a Pico into a Raspberry Pi
2. Build `hx9-control/pod-operation` with `cargo build --release`, ensuring that it is being compiled to the `aarch64` (Raspberry Pi) architecture
3. Copy the executable located in `target/aarch64-unknown-linux-gnu/release/pod-operation` to the Raspberry Pi
4. Run the `pod-operation` executable and observe the waveform on the oscilloscope or connect an LED to the pin outputting the waveform and observe it blinking

samderanova committed:
Add inverter board abstraction

taesungh commented on pod-operation/src/components/inverter_board.rs:
```diff
@@ -0,0 +1,18 @@
+use rppal::uart::{Parity, Uart};
+pub struct InverterBoard {
```
Chore: could we include an empty line after the imports

taesungh commented on pod-operation/src/components/inverter_board.rs:
```diff
@@ -0,0 +1,18 @@
+}
+
+impl InverterBoard {
+	pub fn new() -> Self {
+		let uart = Uart::with_path("/dev/ttyACM0", 9600, Parity::None, 8, 1).unwrap();
```
Chore: although the intent is mostly assumed for a standard [8-N-1](https://en.wikipedia.org/wiki/8-N-1), for improved clarity, could we extract each of these arguments to named global constants.

taesungh commented on pod-operation/src/components/inverter_board.rs:
```diff
@@ -0,0 +1,18 @@
+	}
+
+	pub fn send_control(&mut self, velocity: f32, throttle: f32) {
+		let vel_str = velocity.to_string();
+		let throttle_str = throttle.to_string();
+		let data = vel_str + " " + &throttle_str + "\n";
```
Thought: I think it might be simpler to construct the message string using the [`format!`](https://doc.rust-lang.org/std/macro.format.html) macro rather than concatenating.

taesungh commented on pod-operation/src/components/inverter_board.rs:
```diff
@@ -0,0 +1,18 @@
+		Self { uart }
+	}
+
+	pub fn send_control(&mut self, velocity: f32, throttle: f32) {
```
Chore: could we include a doc comment describing how this encodes and sends the message to the Pi Pico

taesungh replied:
[Doc comments](https://doc.rust-lang.org/reference/comments.html#doc-comments) must begin with _three_ slashes.
Nitpick, compound adjective should be hyphenated (i.e. "space-separated")

taesungh reviewed (commented):
Looks good functionally as verified in lab; minor organizational notes.

samderanova committed:
InverterBoard refactoring

- Extract arguments to Uart::with_path to constants
- Simplify send_control to use format! rather than string concatenation

taesungh reviewed (approved):
Still unsure if we're having a single or multiple Picos, but works for now.

ryescholin reviewed (approved):
Looks great, does this need any functionality in FSM?

samderanova commented:
> Looks great, does this need any functionality in FSM?

Eventually, yes. We'll want to be able to control velocity and throttle from the FSM. In the meantime, though, @taesungh and I will be working on [inverter-control#37](https://github.com/UCI-HyperXite/inverter-control/issues/37) to improve the communication between the Pi and Pico(s).
